src/cli/self_update/windows.rs

```diff
@@ -144,6 +144,11 @@ pub fn do_add_to_path() -> Result<()> {
     _apply_new_path(new_path)
 }
 
+// Nothing to do for Windows for now
+pub fn do_write_env_files() -> Result<()> {
+    Ok(())
+}
+
```

Comment:
This is a literally useless function i.e. it does nothing except further enmesh the two OS' APIs.

I submitted 3 PRs with the express intention in each one of reducing the frequency of these in the codebase so that Linux and Windows devs would not have to consult enmeshed OS APIs, and I was hardly given even 24 hours to review this, whereas at least one of my PRs was held up for weeks in review for not being pretty enough or not using APIs that had no explanation for their usage.

Reply:
This is a fair assessment.  It does increase coupling in an API you worked very hard to decouple.  I can only apologise for the timing and the presentation of this change, I was working under (perhaps imagined) pressure to get a fix out for users because 1.23.0 broke a behaviour people were reasonably relying on and it had broken a number of CIs (not just this, but also the macos 10.13 problem).

I will work on a PR to decouple this again, I apologise for rushing through something less than ideal.